core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeAstBuilder.scala

```diff
@@ -56,31 +55,63 @@ class SnowflakeAstBuilder extends SnowflakeParserBaseVisitor[TreeNode] {
   }
 }
 
-class SnowflakeRelationBuilder extends SnowflakeParserBaseVisitor[Relation] {
-  override def visitObject_ref(ctx: SnowflakeParser.Object_refContext): Relation = {
+class SnowflakeRelationBuilder extends SnowflakeParserBaseVisitor[ir.Relation] {
+  override def visitObject_ref(ctx: SnowflakeParser.Object_refContext): ir.Relation = {
     val tableName = ctx.object_name().id_(0).getText
-    NamedTable(tableName, Map.empty, is_streaming = false)
+    ir.NamedTable(tableName, Map.empty, is_streaming = false)
   }
+
+  override def visitTable_source_item_joined(ctx: Table_source_item_joinedContext): ir.Relation = {
+
+    def translateJoinType(joinType: SnowflakeParser.Join_typeContext): ir.JoinType = {
+      if (joinType == null || joinType.outer_join() == null) {
+        ir.InnerJoin
```

Comment:
can we switch to pattern matching, much easier to manage and read.

Reply:
I'm afraid we can't (although I totally agree it would be better expressed with pattern-matching).

The (Java) classes generated from the grammar (such as `Join_typeContext` here) do not come with a companion object providing an `unapply` method (which what Scala's pattern-matching relies upon). They aren't even data classes (they have no field, only methods). So I don't see any way around writing this kind of (arguably ugly) if-else-chains-involving-null-checks.

Reply:
Does it make sense to define a function like below, and use it in the if-else conditions: 

```
def isInnerJoin(joinType): bool 
  joinType == null || joinType.outer_join() == null

```  

```
  if (isInnerJoin(joinType)):  ir.InnerJoin
  else if(isLeftJoin(joinType)): ir.LeftJoin
   ......
    else: ..... 
```